src/connectedvmware/azext_connectedvmware/custom.py

```diff
@@ -765,6 +768,146 @@ def get_hcrp_machine_id(
+            cmd, rg_name, VMWARE_NAMESPACE, VCENTER_RESOURCE_TYPE, vcenter)
+    assert vcenter_id is not None
+    if resource_name is not None:
+        return create_vm(
+            cmd,
+            client,
+            resource_group_name=rg_name,
+            resource_name=resource_name,
+            vcenter=vcenter,
+            no_wait=no_wait,
+        )
+    
+    logger = get_logger(__name__)
+    arg_client = cf_resource_graph(cmd.cli_ctx)
+
+    vcenter_sub = vcenter_id.split("/")[2]
+    resources_client = get_resources_client(cmd.cli_ctx, vcenter_sub)
+    vcenter = resources_client.get_by_id(vcenter_id, VCENTER_KIND_GET_API_VERSION)
+    logger.info(f"Searching for machines in the vCenter {vcenter.name}...")
+
+    query = f"""
+Resources
+| where type =~ 'Microsoft.HybridCompute/machines'
+| where kind =~ "{vcenter.kind}" or isempty(kind)
+| extend p=parse_json(properties)
+| extend u = tolower(tostring(p['vmUuid']))
+| where isnotempty(u)
+| where location =~ '{vcenter.location}'
+| extend vmUuidRev = strcat(
+    substring(u, 6, 2), substring(u, 4, 2), substring(u, 2, 2), substring(u, 0, 2), '-',
+    substring(u, 11, 2), substring(u, 9, 2), '-',
+    substring(u, 16, 2), substring(u, 14, 2), '-',
+    substring(u, 19))
+| project machineId=id, name, vmUuidRev
+| join kind=inner (
+ConnectedVMwareVsphereResources
+| where type =~ 'Microsoft.ConnectedVMwareVsphere/VCenters/InventoryItems' 
+| where kind =~ 'VirtualMachine'
+| where id startswith '{vcenter.id}/InventoryItems'
+| extend p=parse_json(properties)
+| extend biosId = tolower(tostring(p['smbiosUuid']))
+| extend managedResourceId=tolower(tostring(p['managedResourceId']))
+| project inventoryId=id, biosId, managedResourceId 
+) on $left.vmUuidRev == $right.biosId
+| project-away vmUuidRev
+"""
+    print(query)
+    from time import sleep; sleep(10)
+    
+    # https://github.com/wpbrown/azmeta-libs/blob/4495d2d55f052032fe11416f5c59e2f2e79c2d73/azmeta/src/azmeta/access/resource_graph.py
+    skip_token = None
+    vm_list = []
+    while True:
+        query_options = QueryRequestOptions(skip_token=skip_token)
+        query_request = QueryRequest(
+            subscriptions=[vcenter_sub],
+            query=query,
+            options=query_options,
+        )
+        query_response = arg_client.resources(query_request)
+        if not isinstance(query_response.data, list):
+            query_response.data = [query_response.data]
+        vm_list.extend(query_response.data)
+        skip_token = query_response.skip_token
+        if skip_token is None:
+            break
+    logger.info(f"{len(vm_list)} machines will be attempted to be linked to the vCenter {vcenter.name}...")
+    failed = 0
+    for i, vm in enumerate(vm_list):
+        prefix = f"[{i+1}/{len(vm_list)}]"
+        machineId = vm["machineId"]
+        machineName = vm["name"]
+        inventoryId = vm["inventoryId"]
+        managedResourceId = vm["managedResourceId"]
```

Comment:
If managedResourceId is already present, we can assert  it is matching what we also found from the ARG query.

Reply:
Am checking for managedResourceId if already present now.